### DISCUSSION TITLE
Logs in default data folder

### DISCUSSION THREAD

**Discussion by Trakeur** [DependencyTrack/dependency-track#2902] [Q&A] [answered]
Hi,
Even if I specify a specific path in application.properties for the data folder, the logs files are still saved in the default ~/.dependencytrack path, even though the rest of the data is in the specified folder. 

Is there any way to avoid this?

**Comment by nscuro** [comment]
True, the log file location is configured separately via the [logback configuration](https://logback.qos.ch/manual/configuration.html) file, e.g.: 

https://github.com/DependencyTrack/dependency-track/blob/f40fc2f2805093e57b324fa0f70df09117a6d86d/src/main/resources/logback.xml#L5

This file is used per default, but can be overridden via the `-Dlogback.configurationFile` JVM option. 

What you can do is:

1. Copy the `logback.xml` file and customize as needed
2. For container deployments, mount your customized file into the container
3. Use the `EXTRA_JAVA_OPTIONS` environment variable to pass the path to the customized logback config to DT:
https://github.com/DependencyTrack/dependency-track/blob/f40fc2f2805093e57b324fa0f70df09117a6d86d/src/main/docker/Dockerfile#L20-L21

To test:

```shell
docker run -d \
    -v "$(pwd)/my-logback.xml:/my-logback.xml:ro" \
    -e "EXTRA_JAVA_OPTIONS=-Dlogback.configurationFile=/my-logback.xml" \
    # (all of your usual options and flags)
    dependencytrack/apiserver:4.8.1
```

**Reply by Trakeur** [reply]
Thank you, it worked well. 
However, I don't use docker and start the server with a java command. Before this change I was seeing the logs in real time in the terminal which is quite handy, now nothing. I guess you have to specify the new log path somewhere but I don't know where.

**Reply by nscuro** [reply, marked as answer]
Did you modify anything else other than the file paths?

The config also includes appenders that log directly to standard output

https://github.com/DependencyTrack/dependency-track/blob/f40fc2f2805093e57b324fa0f70df09117a6d86d/src/main/resources/logback.xml#L19-L23

Separately, try adding 

```xml
<appender-ref ref="STDOUT" />
```

to the `<logger` segments, e.g.:

```xml
<logger name="org.dependencytrack" level="${dependencyTrack.logging.level:-INFO}" additivity="false">
  <appender-ref ref="FILE" />
  <appender-ref ref="SECURITY_CONSOLE" />
  <appender-ref ref="STDOUT" />
</logger>
```

**Reply by Trakeur** [reply]
I add the <appender-ref ref="STDOUT" /> to each logger segment and it worked. Thank you !
